configure

```diff
@@ -1062,6 +1067,28 @@ def configure_static(o):
         o['libraries'] += ['-static-libasan']
 
 
+def configure_gold(o):
+  try:
+    proc = subprocess.Popen(shlex.split('ld.gold') + ['-v'], stdin=subprocess.PIPE, 
+                            stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+    match = re.match(r"(GNU gold) .* ([0-9]\.[0-9]+)",
+                         proc.communicate()[0])
+    if match:
+      gold_version = match.group(2)
+      if gold_version > '1.1':
+        o['variables']['gold_linker'] = 'true'
+        if options.section_reordering:
+          o['variables']['goldl_function_reorder'] = options.section_reordering
+        else:
+          o['variables']['goldl_function_reorder'] = os.path.realpath('tools/gold_linker_section_reordering.txt')
+        if(flavor in  ('linux', 'freebsd', 'openbsd')):
```

Comment:
Superfluous parens and can you make sure lines are <= 80 columns?

Also, why is this restricted to Linux and two BSDs?  Isn't the presence of the gold linker enough?

Reply:
Do you think it would make sense to keep this check if `cross_compiling` is defined  ?
`ld.gold` is only supported on these platforms and available by default when `bin.utils` package is installed.

Reply:
If there is a `ld.gold` for the target, then logically it supports function reordering, doesn't it?

I've been thinking it over but I can't think of a good reason to restrict it.